sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/azure/azure.yaml

```diff
@@ -0,0 +1,77 @@
+provider:
+  metadata:
+    name: Azure 
+    category: Cloud provider
+    references:
+      - description: OWASP Top 10 2021 Category A7 - Identification and Authentication Failures
+        link:  https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/
+        type: Standards
+      - description: docs.microsoft.com - Manage storage account access keys
+        link:  https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal
+        type: Documentation
+      - description: OWASP Top 10 2017 Category A3 - Sensitive Data Exposure
+        link:  https://www.owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure
+        type: Standards
+      - description: MITRE, CWE-798 - Use of Hard-coded Credentials
+        link:  https://cwe.mitre.org/data/definitions/798
+        type: Standards
+      - description: MITRE, CWE-259 - Use of Hard-coded Password
+        link:  https://cwe.mitre.org/data/definitions/259
+        type: Standards
+      - description: SANS Top 25 - Porous Defenses
+        link:  https://www.sans.org/top25-software-errors/#cat3
+        type: Standards
+    message: Make sure this Azure Storage Account Key gets revoked, changed, and removed from the code.
+  detection:
+    post:
+        patternNot: "EXAMPLE"
+
+  rules:
+    - id: azure-storage-acount-key
+      metadata:
+        name: Azure Storage Account Keys should not be disclosed
+      examples:
+        - text: >
+            async function main() {
+              const account = process.env.ACCOUNT_NAME || "accountname";
+              const accountKey = process.env.ACCOUNT_KEY || "4dVw+l0W8My+FwuZ08dWXn+gHxcmBtS7esLAQSrm6/Om3jeyUKKGMkfAh38kWZlItThQYsg31v23A0w/uVP4pg==";
+              const sharedKeyCredential = new StorageSharedKeyCredential(account, accountKey);
+              const blobServiceClient = new BlobServiceClient(
+                `https://${account}.blob.core.windows.net`,
+                sharedKeyCredential
+              );
+            }
+          containsSecret: true
+          match: 4dVw+l0W8My+FwuZ08dWXn+gHxcmBtS7esLAQSrm6/Om3jeyUKKGMkfAh38kWZlItThQYsg31v23A0w/uVP4pg==
+        - text: >
+            AccountKey = "BtS7esLAQSrm6/Om3jeyUKKGMkfAh38kWZlItThQYsg31v23A0w/uVP4pg==";
+          containsSecret: false
+        - text: >
+            const connStr = "DefaultEndpointsProtocol=https;AccountName=testaccountname;AccountKey=4dVw+l0W8My+FwuZ08dWXn+gHxcmBtS7esLAQSrm6/Om3jeyUKKGMkfAh38kWZlItThQYsg31v23A0w/uVP4pg==;EndpointSuffix=core.windows.net";
+          containsSecret: true
+          match: 4dVw+l0W8My+FwuZ08dWXn+gHxcmBtS7esLAQSrm6/Om3jeyUKKGMkfAh38kWZlItThQYsg31v23A0w/uVP4pg==
+      detection:
+        pre:
+          include:
+            content:
+              - "core.windows.net"
+        matching:
+          pattern: "['\"`]?([a-zA-Z0-9\/\\+]{86}==)['\"`]?"
```

Comment:
Unfortunately, adding question marks is not an option here:
`['\"]?([a-zA-Z0-9\/\\+]{86}==)['\"]?` is the same as `([a-zA-Z0-9\/\\+]{86}==)` which would result in way too many false positives.
I would stick to the original and accept that we might miss some secrets. It is just not a great secret to detect by its pattern.

Reply:
```suggestion
          pattern: "['\"`]([a-zA-Z0-9\/\\+]{86}==)['\"`]"
```

Reply:
I actually added it to the next regex, should work